packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/base/data_sources/remote/interceptors/auth_interceptor.dart

```diff
@@ -36,51 +43,30 @@ class AuthInterceptor extends Interceptor {
     if (accessToken != null) {
       options.headers['Authorization'] = 'Bearer $accessToken';
     }
-    // TODO: Add your logic here
     super.onRequest(options, handler);
   }
 
   @override
   Future<void> onError(DioError err, ErrorInterceptorHandler handler) async {
     if (err.response?.statusCode == 401) {
-      final Map<String, dynamic> errData = jsonDecode(err.response.toString());
-      if (errData['details'] != 'Bad credentials') {
-        try {
-          await loadNewToken();
-        } on Exception catch (e) {
-          print(e.toString());
-          await logout();
-        }
+      final newToken = await loadNewToken();
+      if (newToken == null) {
+        await _logoutUseCase.execute();
       }
     }
     super.onError(err, handler);
   }
 
-  Future<void> loadNewToken() async {
+  Future<String?> loadNewToken() async {
```

Comment:
Looking at this piece of code, why not make it part of a `FetchAuthTokenUseCase` and use it in this interceptor. This way we make this code reusable elsewhere and remove the need for `getToken` and `getRefreshToken` methods in this `AuthInterceptor`.

Reply:
Sure. I was wondering if this is a good idea.